Extension/src/LanguageServer/extension.ts

```diff
@@ -174,7 +174,9 @@ export async function activate(): Promise<void> {
 
     console.log("starting language server");
     clients = new ClientCollection();
-    ui = getUI();
+    getActiveClient().awaitUntilLanguageClientReady().then(async () =>
```

Comment:
Is there a reason this needs to be deferred until after the language client is ready?  Looking at the implementation of getUI, couldn't you just await this here?  Also, updating the use of `ui` below to `ui?` is always going to fail, as `ui` will always be undefined there.  So, deferring `getUI` means that call to activeDocumentChanged will never occur.

Reply:
It was necessary in order to for the experimentation API to get activated.

Reply:
In general, the need for a promise is because of the experiment so once that is done getUI will return to a normal function. Unless there is a way to use the experimentation api without a promise

Reply:
Colen is asking why we need to wait for the client to be ready before we can get the experimentation service.  Can we do this:
```
    ui = await getUI();
```
instead of:
```
    getActiveClient().awaitUntilLanguageClientReady().then(async () =>
        ui = await getUI()
    );
```

Reply:
The experimentation service doesn't look like it has any dependencies on the client becoming active.

Reply:
Also, if you set `ui` before creating `ClientCollection`, I think there is no risk to assuming the UI will never be `undefined` and the typing could be updated.

Reply:
My mistake, i got the error wrong. what I meant was without awaitUntilLanguageClientReady getUI() will be called multiple times before it can set ui. Which causes multiple constructor calls and resulting in a crash

Reply:
That's probably because ClientCollection creates a Client whose constructor calls `getUI` asynchronously.  If you create the UI before the ClientCollection, I think that should resolve the problem.

Reply:
There appears to be a possible race in `getUI` where multiple callers could enter the `if (!ui)` branch, wait on the experimentation service promise, and then both try to create and set `ui`.  Changing the order of the calls as I suggest above mitigates that, but we may also want to follow up to this PR with a new one that fixes the reentrancy problem in `getUI`.

Reply:
Addressing it like how we discussed